iam.tf

```diff
@@ -206,3 +206,32 @@ resource "aws_iam_policy_attachment" "geff_lambda_vpc_policy_attachment" {
   roles      = [aws_iam_role.geff_lambda_assume_role.name]
   policy_arn = data.aws_iam_policy.geff_lambda_vpc_policy[0].arn
 }
+
+# -----------------------------------------------------------------------------
+# 4. Policy for the DynamoDB table to be used as a backend for batch locking
+# -----------------------------------------------------------------------------
+resource "aws_iam_policy" "dynamodb_table_policy" {
+  count = var.create_dynamodb_table || var.user_managed_dynamodb_table_name != null ? 1 : 0
```

Comment:
can this be simplified to `local.dynamodb_table`? it seems that would make it a clear connection to the `Resource` value below

Reply:
It can, but the target needs to be specified on each apply. 
>  The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work around this, use the -target argument to first apply only the resources that the count depends on.

So this seemed simpler than that.

Reply:
The explanation I got for this was that tf needs the `count` value during planning to know exactly how many resources are to be created, and that it's unable to do that when a `local` is used in such a condition since it derives its value from attributes of other resources.